SmartDeviceLinkTests/DevAPISpecs/NSMutableDictionary+StoreSpec.m

```diff
@@ -0,0 +1,165 @@
+            expectAction(^{
+                expect([testDictionary sdl_objectsForName:testKey ofClass:[NSString class]]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding a custom object using JSON", ^{
+        __block NSDictionary *testJson = @{};
+        __block TestObject *testObject = [[TestObject alloc] initWithDictionary:testJson];
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testJson forName:testKey];
+        });
+
+        it(@"should return object", ^{
+            expect([testDictionary sdl_objectForName:testKey ofClass:[TestObject class]]).to(equal(testObject));
+        });
+        it(@"should raise exceprion", ^{
+            expectAction(^{
+                expect([testDictionary sdl_objectForName:testKey ofClass:[NSNumber class]]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding an array of strings", ^{
+        __block NSString *testObject = @"Object";
+        __block NSArray<NSString *> *testObjectArray = @[testObject];
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testObjectArray forName:testKey];
+        });
+
+        expectAction(^{
+            expect([testDictionary sdl_objectForName:testKey ofClass:[NSString class]]).to(equal(testObject));
+        }).to(raiseException());
+        it(@"should return array", ^{
+            expect([testDictionary sdl_objectsForName:testKey ofClass:[NSString class]]).to(equal(testObjectArray));
+        });
+        it(@"should raise exceprion", ^{
+            expectAction(^{
+                expect([testDictionary sdl_objectForName:testKey ofClass:[NSNumber class]]).to(beNil());
+                expect([testDictionary sdl_objectsForName:testKey ofClass:[NSNumber class]]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding an array of custom objects", ^{
+        __block NSDictionary *testJson = @{};
+        __block TestObject *testObject = [[TestObject alloc] initWithDictionary:testJson];
+        __block NSArray<NSDictionary *> *testJsonArray = @[testJson];
+        __block NSArray<TestObject *> *testObjectArray = @[testObject];
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testJsonArray forName:testKey];
+        });
+
+        it(@"should return array", ^{
+            expect([testDictionary sdl_objectsForName:testKey ofClass:[TestObject class]]).to(equal(testObjectArray));
+        });
+        it(@"should raise exceprion", ^{
+            expectAction(^{
+                expect([testDictionary sdl_objectsForName:testKey ofClass:[NSNumber class]]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding a SDLEnum", ^{
+        __block SDLEnum testObject = @"Object";
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testObject forName:testKey];
+        });
+
+        it(@"should return SDLEnum", ^{
```

Comment:
@"should return correctly when retrieved correctly"

Reply:
fixed in 6f291ca